### PR TITLE
Feature/mentee search

### PR DESCRIPTION
**Pull request by stackpearson**
### Adding search paramaters to the search bar
This PR enabled scrolling through mentees in the desktop view and focused on giving headmasters the ability to search for mentees using any of the following filters:

- Name
- Birthday
- Email
- Timezone
- Minimum English Grade
- Minimum Math Grade
- Minimum Reading Grade
- Minimum School Grade

image

### CODE REVIEW

File: src/components/pages/Headmaster/Mentees/Mentees.js [hunk 1]
```diff
@@ -27,16 +28,67 @@ const Mentees = props => {
...
+    menteesSelection = menteesSelection.filter(item =>
+      item.email.toLowerCase().includes(search.toLowerCase())
+    );
+  } else if (Array.isArray(menteesSelection) && searchBy == 'Timezone') {
+    menteesSelection = menteesSelection.filter(item =>
+      item.availability.time_zone.toLowerCase().includes(search.toLowerCase())
+    );
+    // this dynamically filters grades by the selected class. The 'searchBy' strings must match the corresonding value on the selectBy options as we're just slicing the string as it's passed in.
+  } else if (
+    Array.isArray(menteesSelection) &&
```
Comment (abdimohamud) [inline src/components/pages/Headmaster/Mentees/Mentees.js:81]: I like the way you approached the filtering by input and dropdown, easier to make a copy of the data with a .filter then waiting for the db to query a result back in time

**Review by abdimohamud — APPROVED** [review]
Looks good to me! I'm going to have to pull your changes since I'm working on that component and will make a PR soon with the edit button

**Review by tompsherman — APPROVED** [review]
I really appreciate the granular nature of your commits, getting in small, clearly defined chunks of code /steps that work then building upon them. 

Also i love the the combined dropdown+search method, it is a great idea for creating a user friendly environment

File: src/components/pages/Headmaster/Mentees/Mentees.js (hunk 1 above)
Comment (stackpearson) [reply src/components/pages/Headmaster/Mentees/Mentees.js:81]: Thanks, figured that's easier since we don't know what the true backend looks like anyways.

**Pull request merged**